libjsonutils/cmake/FindRapidJSON.cmake

```diff
@@ -1,46 +1,55 @@
 #FindRapidJSON.cmake
-# 
+#
 # Finds the rapidjson library
 #
+# from http://rapidjson.org/
+#
 # This will define the following variables
 #
-#    RapidJSON_FOUND
-#    RapidJSON_INCLUDE_DIRS
+#    RAPIDJSON_FOUND
+#    RAPIDJSON_VERSION
+#    RAPIDJSON_INCLUDE_DIRS
 #
 # and the following imported targets
 #
 #     RapidJSON::RapidJSON
 #
 # Author: Pablo Arias - pabloariasal@gmail.com
+#
+# clausklein$ cmakelint *.cmake
+# FindRapidJSON.cmake:0: Find modules should use uppercase names;
+#   consider using FindRAPIDJSON.cmake [convention/filename]
+# Total Errors: 1
 
-find_package(PkgConfig)
+find_package(PkgConfig REQUIRED)
 pkg_check_modules(PC_RapidJSON QUIET RapidJSON)
 
-find_path(RapidJSON_INCLUDE_DIR
+find_path(RAPIDJSON_INCLUDE_DIR
     NAMES rapidjson.h
     PATHS ${PC_RapidJSON_INCLUDE_DIRS}
     PATH_SUFFIXES rapidjson
 )
 
-set(RapidJSON_VERSION ${PC_RapidJSON_VERSION})
+set(RAPIDJSON_VERSION ${PC_RapidJSON_VERSION})
 
-mark_as_advanced(RapidJSON_FOUND RapidJSON_INCLUDE_DIR RapidJSON_VERSION)
+# NOTE: (cmakelint) All vars should be uppercase! CK
+mark_as_advanced(RAPIDJSON_FOUND RAPIDJSON_INCLUDE_DIR RAPIDJSON_VERSION)
 
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(RapidJSON
-    REQUIRED_VARS RapidJSON_INCLUDE_DIR
-    VERSION_VAR RapidJSON_VERSION
+find_package_handle_standard_args(RAPIDJSON
+    REQUIRED_VARS RAPIDJSON_INCLUDE_DIR
+    VERSION_VAR RAPIDJSON_VERSION
 )
 
-if(RapidJSON_FOUND)
+if(RAPIDJSON_FOUND)
     #Set include dirs to parent, to enable includes like #include <rapidjson/document.h>
-    get_filename_component(RapidJSON_INCLUDE_DIRS ${RapidJSON_INCLUDE_DIR} DIRECTORY)
+    get_filename_component(RAPIDJSON_INCLUDE_DIRS ${RAPIDJSON_INCLUDE_DIR} DIRECTORY)
```

Comment:
Better is to just do `find_path(NAMES rapidjson/rapidjson.h)` which will get you the parent directory automatically.

Reply:
yes, we already did `find_path` above to find rapidjson's include directory, this gives you `/usr/linclude/rapidjson`, which is correct, the problem is that we can't do `#include <rapidjson/document.h>` in the code, but only `#include <document.h>`, which a little inconvenient as the reader doesn't have the library name when exporting. So here we set the actual `RapidJSON_INCLUDE_DIRS` to one level above: `/usr/include`